tools/src/podio-dump-tool.cpp

```diff
@@ -0,0 +1,267 @@
+  --dump-edm DUMP_EDM   Dump the specified EDM definition from the file in yaml format
+  --version             show program's version number and exit
+)";
+
+void printUsageAndExit() {
+  fmt::print(stderr, "{}\n", usageMsg);
+  std::exit(1);
+}
+
+auto getArgumentValueOrExit(const std::vector<std::string>& argv, std::vector<std::string>::const_iterator it) {
+  const int argc = argv.size();
+  const auto index = std::distance(argv.begin(), it);
+  if (index > argc - 2) {
+    printUsageAndExit();
+  }
+  return argv[index + 1];
+}
+
+std::vector<size_t> parseEventRange(const std::string& evtRange) {
+  const auto splitRange = splitString(evtRange, ',');
+  const auto parseError = [&evtRange]() {
+    fmt::print(stderr, "'{}' canot be parsed into a list of entries\n", evtRange);
+    std::exit(1);
+  };
+
+  if (splitRange.size() == 1) {
+    const auto colonSplitRange = splitString(evtRange, ':');
+    if (colonSplitRange.size() == 1) {
+      return {parseSizeOrExit(splitRange[0])};
+    } else if (colonSplitRange.size() == 2) {
+      // we have two numbers signifying an inclusive range
+      const auto start = parseSizeOrExit(colonSplitRange[0]);
+      const auto end = parseSizeOrExit(colonSplitRange[1]);
+      std::vector<size_t> events(end - start + 1);
+      std::iota(events.begin(), events.end(), start);
+      return events;
+    } else {
+      parseError();
+    }
+  } else {
+    std::vector<size_t> events;
+    events.reserve(splitRange.size());
+    std::ranges::transform(splitRange, std::back_inserter(events),
+                           [](const auto& elem) { return parseSizeOrExit(elem); });
+
+    return events;
+  }
+
+  parseError();
+  return {};
+}
+
+ParsedArgs parseArgs(std::vector<std::string> argv) {
+  // find help or version
+  if (const auto it = findFlags(argv, "-h", "--help", "--version"); it != argv.end()) {
+    if (*it == "--version") {
+      fmt::print("podio {}\n", podio::version::build_version);
+    } else {
+      fmt::print("{}\n{}", usageMsg, helpMsg);
+    }
+    std::exit(0);
+  }
+
+  ParsedArgs args;
+  // detailed flag
+  if (const auto it = findFlags(argv, "-d", "--detailed"); it != argv.end()) {
+    args.detailed = true;
+    argv.erase(it);
+  }
+  // category
+  if (const auto it = findFlags(argv, "-c", "--category"); it != argv.end()) {
+    args.category = getArgumentValueOrExit(argv, it);
+    argv.erase(it, it + 2);
+  }
+  // event range
+  if (const auto it = findFlags(argv, "-e", "--entries"); it != argv.end()) {
+    args.events = parseEventRange(*(it + 1));
+    argv.erase(it, it + 2);
+  }
+  // dump-edm
+  if (const auto it = findFlags(argv, "--dump-edm"); it != argv.end()) {
+    args.dumpEDM = getArgumentValueOrExit(argv, it);
+    argv.erase(it, it + 2);
+  }
+
+  if (argv.size() != 1) {
+    printUsageAndExit();
+  }
+  args.inputFile = argv[0];
+
+  return args;
+}
+
+template <typename T>
+std::string getTypeString() {
```

Comment:
```suggestion
constexpr std::string getTypeString() {
```
I'm not sure this will change anything...

Reply:
Made it a `consteval std::string_view`, so it's definitely evaluated at compile time.